api/tests/Feature/SnapshotTest.php

```diff
@@ -46,14 +46,23 @@ public function testCreateSnapshot()
             ->asApplicant()
             ->create();
 
-        $poolCandidate = PoolCandidate::factory()->create([
-            "user_id" => $user->id,
-            "pool_candidate_status" => ApiEnums::CANDIDATE_STATUS_DRAFT
-        ]);
-        $poolCandidateUnrelated = PoolCandidate::factory()->create([
-            "user_id" => $user->id,
-            "pool_candidate_status" => ApiEnums::CANDIDATE_STATUS_DRAFT
-        ]);
+        $pool1 = Pool::factory()->published()->create();
+        $pool2 = Pool::factory()->published()->create();
+
+        $poolCandidate = PoolCandidate::factory()
+            ->availableInSearch()
+            ->create([
+                "user_id" => $user->id,
+                "pool_id" => $pool1->id,
+                "pool_candidate_status" => ApiEnums::CANDIDATE_STATUS_DRAFT
+            ]);
+        $poolCandidateUnrelated = PoolCandidate::factory()
+            ->availableInSearch()
+            ->create([
+                "user_id" => $user->id,
+                "pool_id" => $pool2->id,
+                "pool_candidate_status" => ApiEnums::CANDIDATE_STATUS_DRAFT
```

Comment:
This creates an odd state.
`availableInSearch()` effectively submits it and then it is being reverted to draft. 
I don't think it impacts testing, it just creates a model state that is/should be impossible.

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/e7b911a0-adcc-455f-98e5-83831aba6ef6)

Reply:
Oh wait, hang on, should `availableInSearch` also defined a value for `submitted_at`? I think I've wandered into spinoff territory

Reply:
Oh, you're right.  :facepalm: We don't have a factory state for draft yet but I see the factory has some conditional logic when the pool_candidate_status is DRAFT so I'll leave it there.
Ref: [remove pool candidate factory states](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7243/commits/574de73879e4f96029aeb2178c43890c0d8e90c7)